spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -1126,27 +1129,36 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
               val cometMessage =
                 if (cometException.getCause != null) cometException.getCause.getMessage
                 else cometException.getMessage
-              if (CometSparkSessionExtensions.isSpark40Plus) {
-                // for Spark 4 we expect to sparkException carries the message
+              // for comet decimal conversion throws ArrowError(string) from arrow - across spark versions the message dont match.
+              if (sparkMessage.contains("cannot be represented as")) {
                 assert(
-                  sparkException.getMessage
-                    .replace(".WITH_SUGGESTION] ", "]")
-                    .startsWith(cometMessage))
-              } else if (CometSparkSessionExtensions.isSpark34Plus) {
-                // for Spark 3.4 we expect to reproduce the error message exactly
-                assert(cometMessage == sparkMessage)
+                  cometMessage.contains("cannot be represented as") || cometMessage.contains(
+                    "too large to store"))
               } else {
```

Comment:
There are message modifications below per spark version
Would you mind update them instead of creating another if branch?